### PR TITLE
Add compatible go 1.11 and go 1.12

### PR DESCRIPTION
**Pull request by edwardpro**
Feature 1:
To resolve goproxy can't compile under go 1.12.x.
issue: https://github.com/goproxyio/goproxy/issues/55
Feature 2:
And add git client version limit to git 2.x.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1 +0,0 @@
-module github.com/goproxyio/goproxy
```
Comment (oiooj) [inline go.mod]: why you remove this line?

**Comment by oiooj** [comment]
Thanks for your help. These commits should be rebased and fix the tests failed in Docker Cloud.

File: go.mod (hunk 1 above)
Comment (edwardpro) [reply go.mod]: It looks like a mistake.
I recover module files.

**Comment by edwardpro** [comment]
> Thanks for your help. These commits should be rebased. And fix the tests in Docker Cloud please.

OK, I had fix docker build problem.

File: main.go [hunk 2]
```diff
@@ -24,6 +26,23 @@ func init() {
...
+	isGitValid := checkGitVersion()
+	if !isGitValid {
+		log.Fatal("Error in git version, please check your git installed in local, must be great 2.0")
+	}
+}
+
+func checkGitVersion() bool {
+	var err error
+	var ret []byte
+	cmd := exec.Command("/usr/bin/git", "version")
```
Comment (oiooj) [inline main.go]: The last question: `/usr/bin/git` seems not right. This just a default path for git.

Comment (edwardpro) [reply main.go]: fix it, changes it to "git version", use the local environment.

**Comment by oiooj** [comment]
Thanks for your patience with this. @edwardpro

**Pull request merged**